### PR TITLE
chore(docs): update securing pages tutorial

### PR DESCRIPTION
**Pull request by ndom91**
## Reasoning 💡

Draft PR to address #3973 and update some tutorials and other documentation to bring `getServerSession` and `middleware` related documentation up to speed.

## Checklist 🧢

- [x] Documentation
- [ ] Tests
- [ ] Ready to be merged

### CODE REVIEW

File: docs/docs/tutorials/securing-pages-and-api-routes.md [hunk 1]
```diff
@@ -40,12 +40,28 @@ export default function Page() {
...
+
+With NextAuth.js 4.2.0 and Next.js 12, you can now protect your pages via the middleware pattern more easily. If you would like to protect all pages, you can create a `_middleware.js` file in your root `pages` directory which looks like this.
+
+```js title="/pages/_middleware.js"
+export { default } from "next-auth/middleware"
+```
+
+Otherwise, if you only want to protect a subset of pages, you could put it in a subdirectory as well, for example in `/pages/admin/_middleware.js` would protect all pages under `/admin`.
+
+This new `withAuth` middleware client only supports the `"jwt"` [session strategy](https://next-auth.js.org/configuration/options#session) at the moment.
```
Comment (lluia) [inline docs/docs/tutorials/securing-pages-and-api-routes.md:53]:
```suggestion
For the time being, the `withAuth` middleware only supports `"jwt"` as [session strategy](https://next-auth.js.org/configuration/options#session).
```

File: apps/dev/pages/api/examples/protected.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 // This is an example of to protect an API route
-import { getSession } from "next-auth/react"
+import { getServerSession } from "next-auth/react"
```
Comment (balazsorban44) [inline apps/dev/pages/api/examples/protected.js:2]: import { getServerSession } from "next-auth/next"

File: apps/dev/pages/api/examples/protected.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
 
 export default async (req, res) => {
-  const session = await getSession({ req })
+  const session = await getServerSession({ req })
```
Comment (balazsorban44) [inline apps/dev/pages/api/examples/protected.js:5]: `const session = await getServerSession({ req, res }, options)`

And we have to decide where options come from.

File: apps/dev/pages/server.js [hunk 4]
```diff
@@ -11,11 +11,11 @@ export default function Page() {
     <Layout>
       <h1>Server Side Rendering</h1>
       <p>
-        This page uses the universal <strong>getSession()</strong> method in{" "}
-        <strong>getServerSideProps()</strong>.
+        This page uses the universal <strong>getServerSession()</strong> method
```
Comment (balazsorban44) [inline apps/dev/pages/server.js:14]: `getServerSession` is not universal, it's server-only. Basically:

Before:
 - getSession: client, server

After:
 - getSession: client
 - getServerSession: server

File: docs/docs/getting-started/client.md [hunk 5]
```diff
@@ -165,11 +165,11 @@ See repository [`README`](https://github.com/nextauthjs/react-query) for more de
 - Client Side: **Yes**
 - Server Side: **Yes**
```
Comment (balazsorban44) [inline docs/docs/getting-started/client.md:166]: This should be changed to No

File: docs/docs/getting-started/client.md [hunk 6]
```diff
@@ -178,23 +178,28 @@ async function myFunction() {
 }
 ```
 
-#### Server Side Example
+Read the tutorial [securing pages and API routes](/tutorials/securing-pages-and-api-routes) to know how to
+fetch the session in server side calls using `getServerSession()`.
+
+## getServerSession()
+
```
Comment (balazsorban44) [inline docs/docs/getting-started/client.md:185]: - Client Side: **No**
- Server Side: **Yes**

**Comment by ndom91** [comment]
@balazsorban44 okay I think your comments have been addressed here. The last thing thats open afaik is deciding where the `options` arg comes from, as you mentioned. Right?

**Comment by ndom91** [comment]
Alright I've updated this PR to remove the `getServerSession` section I'd added to the `client.md` docs page in favor of @ThangHuuVu's awesome PR which adds this (https://github.com/nextauthjs/next-auth/pull/4116). 

This one should be good to go then unless any other changes should be made.

**Review by 0ubbe — APPROVED** [review]
Good from my side 💚

**Comment by ThangHuuVu** [comment]
- I'd suggest revisit this after we finish the API change here: https://github.com/nextauthjs/next-auth/pull/4116 🤔

**Comment by lluia** [comment]
@ThangHuuVu, I agree 👍🏽; I think both PRs should be merged simultaneously, actually. As this one is updating all the references of server-side`getSession()` to `getServerSession()` and yours is adding the docs for `getServerSession()`.

I'll review your PR as well and hopefully, we can move this forward 🍵

**Comment by ThangHuuVu** [comment]
> @ThangHuuVu, I agree 👍🏽; I think both PRs should be merged simultaneously, actually. As this one is updating all the references of server-side`getSession()` to `getServerSession()` and yours is adding the docs for `getServerSession()`.
> 
> I'll review your PR as well and hopefully, we can move this forward 🍵

Sounds good, @lluia! In my PR, I changed the API of `getServerSesison` a bit to match with the API of the default `NextAuth` handler as @balazsorban44 suggested. It would affect the docs/example changes in this PR as well

**Comment by ndom91** [comment]
Yeah im up for whatever, feel free to merge this if yall wanna time it together :+1:

**Comment by kfranqueiro** [comment]
As someone currently looking at upgrading an app from next-auth v3 to v4 who was very confused by things failing until stumbling across https://github.com/nextauthjs/next-auth/issues/4075, and subsequently very confused that this is not documented, evidently due to this PR being in limbo for over 3 months... is there a reason it hasn't been merged and is now in danger of being closed as stale?

**Comment by kfranqueiro** [comment]
Not sure why stalebot closed this when I commented on it? Does that not count as activity?

`getServerSession` is not mentioned at all in the docs currently on the site. This PR looked like it would address that. The fact that this was opened 5 months ago, never merged, and now closed seems like a big problem. How are people actually managing to use v4 successfully if the docs are this behind?

**Comment by ndom91** [comment]
Not sure why stalebot closed this as well. Theres some overlap between this and https://github.com/nextauthjs/next-auth/pull/4116 PR. No excuse as to why it hasn't been merged yet though, `getServerSession` should probably be officially mentioned in the docs at this point :see_no_evil:

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Now that #4116 is merged, could we get this PR up to date? :pray:

**Comment by ndom91** [comment]
> Now that #4116 is merged, could we get this PR up to date? :pray:

Yup, was just wondering if I should still do this one or what. Will update it shortly 👍

**Comment by ndom91** [comment]
@balazsorban44 @ThangHuuVu alright this one is finally updated with `unstable_getServerSession` as well :tada: 

Anything else that you guys would like updated here? Or good to go?

**Review by ThangHuuVu — CHANGES_REQUESTED** [review]
@ndom91 `unstable_getServerSession` now accepts 3 arguments so I suggest the changes accordingly; please add the import for `authOptions` in the example/code as well. This is what I and @balazsorban44 came up with attempting to make this function looks similar to `NextAuthNextHandler`. 💡

**Comment by ndom91** [comment]
@ThangHuuVu great call with the arguments, I didnt realize that at first and was stuck for a minute during my upgrade to `4.6.1` in a sideproject of mine as well :joy:

Appreciate all the recommended diffs btw!

Also updated the text a bit in the SSR recommendation page based on your feedback :+1:

**Review by ThangHuuVu — APPROVED** [review]
Yeah, great work, PR looks ready to be merged now!

**Comment by LOGOLFGODORD** [comment]
Great !! nice work team ! Always there !! yeeeeeee Sent from Mail for Windows From: Thang VuSent: Friday, June 24, 2022 3:50 AMTo: nextauthjs/next-authCc: SubscribedSubject: Re: [nextauthjs/next-auth] chore(docs): update securing pages tutorial (PR ***@***.*** approved this pull request.Yeah, great work, PR looks ready to be merged now!—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Pull request merged**